src/tests/testIoManip.cc

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "base/IoManip.h"
+#include "compat/cppunit.h"
+#include "unitTestMain.h"
+
+#include <cstdint>
+#include <limits>
+#include <sstream>
+
+class TestIoManip: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(TestIoManip);
+    CPPUNIT_TEST(testAsHex);
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+    void testAsHex();
+};
+
+CPPUNIT_TEST_SUITE_REGISTRATION( TestIoManip );
+
+/// resets the given stream, including any formatting
+static void
+resetStream(std::ostringstream &dirty)
+{
+    std::ostringstream clean;
+    dirty.swap(clean);
+}
+
+/// returns the result of printing the given manipulator
+template <typename Integer>
+static std::string
+toStdString(const AsHex<Integer> &manipulator)
+{
+    std::ostringstream os;
+    os << manipulator;
+    return os.str();
+}
+
+void
+TestIoManip::testAsHex()
+{
+    // basic values
+    CPPUNIT_ASSERT_EQUAL(std::string("0"), toStdString(asHex(0)));
+    CPPUNIT_ASSERT_EQUAL(std::string("123abc"), toStdString(asHex(0x123abc)));
+
+    // large values
+    CPPUNIT_ASSERT_EQUAL(std::string("7fffffffffffffff"), toStdString(asHex(std::numeric_limits<int64_t>::max())));
```

Comment:
What about negative values such as `-1` or `std::numeric_limits<int64_t>::min()` ?

Reply:
Negative values are tested in the next blob. Look for `// negative values` title and a comment that explains why we do not support them in a meaningful way.

The same comment implies that testing negative values of unknown (e.g., `-1`) or very large (e.g., `std::numeric_limits<int64_t>::min()`) may not be such a good idea due to (implicit) casting and associated undefined behavior on over/underflows. I would not violently oppose adding those two lines/cases, but I did remove them (after adding them myself) and do not recommend adding them back. 

The primary point here is that std::hex and AsHex do not support negative hex numbers (in a meaningful way). We cannot check for negative numbers and compile time and (currently, in the official code and this PR) do not want to throw at runtime.

Reply:
Ah right. Sorry I missed that.